definitions/npm/styled-components_v3.x.x/flow_v0.53.x-/styled-components_v3.x.x.js

```diff
@@ -0,0 +1,388 @@
+  & $npm$styledComponents$ReactComponentStyledStaticPropsWithComponent<Props, ComponentList>
+  & $npm$styledComponents$ReactComponentIntersection<Props, DefaultProps>
+
+// ---- TAGGED TEMPLATE LITERAL ----
+type $npm$styledComponents$ReactComponentStyledTaggedTemplateLiteral<Props, ComponentList> =
+  & $npm$styledComponents$ReactComponentStyledStaticProps<Props, ComponentList>
+  & $npm$styledComponents$TaggedTemplateLiteral<$npm$styledComponents$ReactComponentStyled<Props, ComponentList>>
+
+type $npm$styledComponents$ReactComponentStyledTaggedTemplateLiteralWithComponent<Props, ComponentList> =
+  & $npm$styledComponents$ReactComponentStyledStaticPropsWithComponent<Props, ComponentList>
+  & $npm$styledComponents$TaggedTemplateLiteral<$npm$styledComponents$ReactComponentStyled<Props, ComponentList>>
+
+// ---- WITHTHEME ----
+type $npm$styledComponents$WithThemeReactComponentClass = <
+  InputProps: { theme: $npm$styledComponents$Theme },
+  InputDefaultProps: {},
+  OutputProps: $Diff<InputProps, { theme: $npm$styledComponents$Theme }>,
+  OutputDefaultProps: InputDefaultProps & { theme: $npm$styledComponents$Theme },
+>($npm$styledComponents$ReactComponentClass<InputProps, InputDefaultProps>) => $npm$styledComponents$ReactComponentClass<OutputProps, OutputDefaultProps>
+
+type $npm$styledComponents$WithThemeReactComponentClassUndefinedDefaultProps = <
+  InputProps: { theme: $npm$styledComponents$Theme },
+  OutputProps: $Diff<InputProps, { theme: $npm$styledComponents$Theme }>,
+>($npm$styledComponents$ReactComponentClassUndefinedDefaultProps<InputProps>) => $npm$styledComponents$ReactComponentClass<OutputProps, { theme: $npm$styledComponents$Theme }>
+
+type $npm$styledComponents$WithThemeReactComponentFunctional = <
+  InputProps: { theme: $npm$styledComponents$Theme },
+  InputDefaultProps: {},
+  OutputProps: $Diff<InputProps, { theme: $npm$styledComponents$Theme }>,
+  OutputDefaultProps: InputDefaultProps & { theme: $npm$styledComponents$Theme },
+>($npm$styledComponents$ReactComponentFunctional<InputProps, InputDefaultProps>) => $npm$styledComponents$ReactComponentFunctional<OutputProps, OutputDefaultProps>
+
+type $npm$styledComponents$WithThemeReactComponentFunctionalUndefinedDefaultProps = <
+  InputProps: { theme: $npm$styledComponents$Theme },
+  OutputProps: $Diff<InputProps, { theme: $npm$styledComponents$Theme }>
+>($npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<InputProps>) => $npm$styledComponents$ReactComponentFunctional<OutputProps, { theme: $npm$styledComponents$Theme }>
+
+type $npm$styledComponents$WithTheme =
+  & $npm$styledComponents$WithThemeReactComponentClass
+  & $npm$styledComponents$WithThemeReactComponentClassUndefinedDefaultProps
+  & $npm$styledComponents$WithThemeReactComponentFunctional
+  & $npm$styledComponents$WithThemeReactComponentFunctionalUndefinedDefaultProps
+
+// ---- MISC ----
+type $npm$styledComponents$Theme = {[key: string]: mixed};
```

Comment:
Can we change this to `Object`? It's not working probably when you use a nested theme.